src/agents/base_agent_component.py

```diff
@@ -109,86 +109,86 @@ def get_agent_summary(self):
-                            f"Exception name: {type(e).__name__}\n"
-                            f"Exception info: {str(e)}\n"
-                            f"Stack trace: {traceback.format_exc()}\n\n"
-                            f"Data: {json.dumps(data)}"
-                        )
-                        log.error(error_message)
-                        action_response = ActionResponse(
-                            message=f"Internal error: {type(e).__name__} - Error invoking action. Details: {str(e)}",
-                            error_info=ErrorInfo(
-                                error_message=error_message,
-                            ),
-                        )
-                else:
-                    log.warning(
-                        "Unauthorized access attempt for action %s. Data: %s",
-                        action_name,
-                        json.dumps(data),
-                    )
-                    action_response = ActionResponse(
-                        message="Unauthorized: You don't have permission to perform this action.",
-                    )
+        action = self.action_list.get_action(action_name)
+        if not action:
+            log.error(
+                "Action not found: %s. Data: %s", action_name, json.dumps(data)
+            )
+            return ActionResponse(
+                message="Internal error: Action not found. Please try again",
+            )
+        return action
 
+    def _resolve_action_parameters(self, params, session_id, data, file_service):
+        try:
+            resolved_params = recursive_file_resolver(
+                params,
+                resolver=file_service.resolve_all_resolvable_urls,
+                session_id=session_id,
+            )
+            return resolved_params, None
+        except Exception as e:
+            log.error(
+                "Error resolving file service URLs: %s. Data: %s",
+                str(e),
+                json.dumps(data),
+                exc_info=True,
+            )
+            return params, ActionResponse(
+                message=f"Error resolving file URLs. Details: {str(e)}",
+            )
+
+    def _execute_action(self, action, resolved_params, user_properties, action_name, data):
+        session_id = user_properties.get("session_id")
+        identity = user_properties.get("identity")
+        middleware_service = MiddlewareService()
+
+        if not middleware_service.get("base_agent_filter")(user_properties, action):
+            log.warning(
+                "Unauthorized access attempt for action %s. Data: %s",
+                action_name,
+                json.dumps(data),
+            )
+            return ActionResponse(
+                message="Unauthorized: You don't have permission to perform this action.",
+            )
+        try:
+            meta = {
+                "session_id": session_id,
+                "identity": identity,
```

Comment:
This is basically the change (adding identity to the meta object) but SonarQube complained about cognitive complexity so I had to refactor the invoke method.